### ISSUE TITLE
Alchemy continues to perform Rank XX Experiments while Research task active

### ISSUE THREAD

**Issue by paladinpizza** [Phr33d0m/NW-Profession-Bot#167]
If you assign more than a single slot for Alchemy tasks, the bot continues to perform Rank X experimentation while Researching to level up, causing consumption of Aqua Regia/Vitae and other resources.  Would it be possible to add a trap/exclusion into the code to not perform Experiments while the Research task is active?  I know that this is more of a professions bug than a bug with the bot, as the profession offers the task even though it shouldn't.

**Comment by noonereally** [comment]
Maybe if the `"Alchemy_Tier3_Experimentation_Rank14"` removed (second item in all the levels). 
Not sure, I never altered the alchemy profile. 
So ideas are welcomed.

`14: ["Alchemy_Tier3_Experiment_Rank15", "Alchemy_Tier3_Experimentation_Rank14", "Alchemy_Tier3_Refine_Basic", "Alchemy_Tier2_Refine_Basic", "Alchemy_Tier1_Refine_Special", "Alchemy_Tier3_Refine_Basic", "Alchemy_Tier3_Gather_Basic"],`

**Comment by dlebedynskyi** [comment]
actually running Experimentation is not a bad idea. You still can get alchemy knowledge out of it. Games does not care if you run experimentation on rank 1 and got 100 knowledge. Game only removed Experimentation  after you have opened all available potions for that level. So running exp while research is on gives you 
- all potions for that level 
  - possible faster leveling due to extra knowledge before next experimentation is available

**Comment by gomsnothereman** [comment]
i had always wondered if running the lower level experiments till you couldn't anymore, BEFORE leveling would be more efficient than just running them till you can do the research then immediately doing the research.. considering the "basic alchemical knowledge" items are used all the way up to getting level 25..

the only thing that sucks is you can ONLY have 1 experiment going on at a time no matter how many slots you have assigned to alchemy..

**Comment by paladinpizza** [comment]
For additional clarification - this is level 20 and up.  Pre-20, yes - once you have performed enough experiments to gain all potions for that level, you typically have gained the number of Alchemical Knowledge required to enable the Research task, and the Experiment task goes away.

However, post level 20 - Alchemy is a bit buggy in the fact that even though you may have already experimented enough to gain both A) all potions, and B) required # Alch Knowledge - the Experimentation task is still available.  You do not gain _any_ additional potions, or additional Alchemical Knowledge at that point.  I have seen a few times where until or unless Research is kicked off you do occasionally get additional Knowledge items, but if memory serves - they disappeared once the research completed and you leveled up.  I'll try to verify that this weekend.

To further clarify, I burnt through approximately 230 Aqua Regia going from level 20 to 25.  It's going to be a few days before I have a toon with alchemy at level 20 to get detailed with it and give exact numbers and reverify.

Hope this helps - I'll try to provide additional supporting hard data as soon as I can, rather than (mostly) anecdotal.  Thanks for your patience, input, and kick ass work.  As soon as things at work cool down a little, I'm going to try and see if my background in Taleo coding and integration transfers over to this stuff and start contributing more than bug reports/requests ;)

**Comment by inv-github** [comment]
Sorry to totally necro, but this was bothering me a bit so I took a quick look at it.  I'm not familiar with js so this is a total hack and I didn't want to put up a pull request for it.  For those that don't mind a total hack, around line 3086 of the current version, near

``` Javascript
// Search for task to start
var task = searchForTask(taskName, prof.taskName, profile, level);
```

I added the following right below:

``` Javascript
if (prof.taskName === "Alchemy" && taskName.includes("Experimentation") && !taskName.includes("Alchemy_Tier3_Experimentation_Rank20") && i > 0) {
    var thisTask = unsafeWindow.client.dataModel.model.craftinglist['craft_' + prof.taskName].entries.filter(function(entry) {
        return entry.def && entry.def.name == list[i - 1];
    })[0];
    if (!thisTask) {
        task = false;
    }
}
```

This assumes that the first task is the research and the 2nd task is the experiment, so it's not very robust.  It will then go to the 3rd task, which is usually making vitriol.  Not the best possible use of time but better than nothing.  Ideally, it would start making other stuff for the next level of experiment but at least vitriol isn't wasted.  I wasn't able to generate any addition knowledge by doing the experiments after starting the research but I didn't track it too closely.
